CHANGELOG.md

```diff
@@ -3,7 +3,134 @@
 All notable changes to this project will be documented in this file. This
 project adheres to [Semantic Versioning](http://semver.org/).
 
-## Unreleased
+## 0.52.0
+
+This release contains some enhancements, bugfixes, and a new builtin function.
+
+### Allow Adding Labels via Discovery
+
+Previously OPA did not allow any updates to the labels provided in the boot configuration via the discovered (ie. service)
+config. This was done to avoid breaking the discovery configuration. But there are use cases where labels can serve as a convenient
+way to pass information that could be used for example in policies, status updates or decision logs. This change allows
+additional labels to be configured in the service config which are then made available during runtime.
+
+See [the Discovery documentation](https://www.openpolicyagent.org/docs/v0.52.0/management-discovery/#limitations)
+for more details.
+
+Authored by @mjungsbluth.
+
+### New Built-In Function: crypto.hmac.equal
+
+`crypto.hmac.equal` provides a convenient way to compare hashes generated by MD5, SHA-1, SHA-256 and SHA-512 hashing algorithms.
+
+Below is a real world example of how this built-in function can be utilized. Imagine our server is registered as a
+GitHub webhook which subscribes to certain events on GitHub.com. Now we want to limit requests to those coming from GitHub.
+One of the ways to do that is to first set up a secret token and validate the information. Once we create the token on GitHub,
+we'll set up an environment variable that stores this token and make it available to OPA via the `opa.runtime` built-in.
+In the case of GitHub webhooks the validation is done by comparing the hash signature received in the `X-Hub-Signature-256`
+header and calculating a hash using the secret token and payload body. The `check_signature` rule implements this logic.
+Using a plain `==` operator is not advised due to security concerns and hence the new `crypto.hmac.equal` built-in function
+leverages [Go's hmac.Equal](https://pkg.go.dev/crypto/hmac#Equal) which compares two MACs for equality without
+leaking timing information. This helps mitigate certain timing attacks against regular equality operators.
```

Comment:
I'm not feeling strongly about this, but I'd perhaps drop these sentences. Proofing Rego code against timing attacks, or even suggesting that if you use this hmac.equal, you'll be safe of such attack vectors doesn't seem like something we should stress 🤔

Reply:
Removed those sentences.